include/modem/location.h

```diff
@@ -267,6 +267,17 @@ struct location_gnss_config {
 	 * @note Only supported with modem firmware v1.3.2 or later.
 	 */
 	bool visibility_detection;
+
+	/**
+	 * @brief Enable GNSS priority mode if GNSS does not get enough runtime due to LTE idle mode
+	 * operations.
+	 *
+	 * @details If set to true, the library triggers GNSS priority mode if five consecutive PVT
+	 * messages indicate that GNSS is blocked by LTE idle mode operations. This is especially
+	 * helpful if A-GPS or P-GPS is not enabled or downloading assistance data fails and GNSS
+	 * module has to decode navigation data from the satellite broadcast.
+	 */
+	bool priority_mode;
```

Comment:
We could add something about the data transfer expectations. Something like:

If data is sent during the priority mode, it will be buffered and sent after the priority mode has finished.
If data would be available for receiving during the priority mode, it could be buffered by the network and sent to the device after the priority mode has finished. However, it is possible that network drops the data, or some protocol timer expires causing data transfer to stop.

@tokangas @jhirsi

Reply:
It's a very nice text, but should it be here instead https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/nrf_modem/doc/gnss_interface.html ?
We could then refer to that document. What do you think @peknis ?

Reply:
Yep, it should be there actually.
I don't know how do we refer to there from the doxygen comments other than adding a link (which may get broken) to the text.

Reply:
I like the idea of moving longer texts to conceptual documentation instead of having it in API docs and then linking to it.

Reply:
GNSS API mentions also that priority mode is max 40s.

Maybe we should just have send/receive data comments and 40s blocking time documented for the priority mode here. Simply because users are looking at location library abstraction and the less they need to follow the links the better. Some duplication of information is not always that bad.

Reply:
Ok, I'll improve the comment here along the lines @trantanen proposed and let @peknis handle any other documents if need be.